### PR TITLE
MHV Landing Page Secondary Nav [bugfix]

### PR DESCRIPTION
**Pull request by radavis**
## Summary

- **Does not render MHV Secondary Nav for unregistered users**

Other Changes/Refactors:

- Moves MHV Landing Page feature toggles to mocks folder (single source of truth)
- Uses `isVAPatient` selector to determine registration status
- Removes hasHealthData in favor of `isVAPatient` selector (duplicate functionality)
- Removes references to facilities, as we don't care about which EHR the user is associated with any longer.
- Removes unnecessary feature toggle fixtures
- Updates e2e test descriptions to reflect domain language
- Removes unused mock-api code

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#83771

## Testing done

- localhost, e2e

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    Screenshot 2024-05-23 at 10-43-38 My HealtheVet Veterans Affairs    |     Screenshot 2024-05-23 at 10-56-07 My HealtheVet Veterans Affairs  |
| Desktop |    Screenshot 2024-05-23 at 10-43-12 My HealtheVet Veterans Affairs    |    Screenshot 2024-05-23 at 10-55-56 My HealtheVet Veterans Affairs   |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

**Review by carlosfelix2 — DISMISSED** [review]

**Review by jzucadi — DISMISSED** [review]

**Review by ataker — DISMISSED** [review]

File: src/applications/mhv-landing-page/tests/e2e/alerts/unregistered.cypress.spec.js [hunk 1]
```diff
@@ -8,13 +8,14 @@ const viewportSizes = ['va-top-desktop-1', 'va-top-mobile-1'];
 const noHealthDataHeading = /You don’t have access to My HealtheVet/i;
 
 describe(appName, () => {
-  describe('Display content based on patient facilities', () => {
+  describe('Display content based on patient registration', () => {
```
Comment (dcloud) [inline src/applications/mhv-landing-page/tests/e2e/alerts/unregistered.cypress.spec.js:11]: The underlying `va_patient` field on `vets-api` just looks at the facility count, so I'm not sure registration is closer to the truth here

**Review by dcloud — DISMISSED** [review]
I'm happy to see the removal of redundant/obsolete logic, the use of better selectors, and the reduction in complexity in mocks/tests.

I think this PR goes beyond the description of a bugfix, so I was spending time trying to think of why some decisions were made in the past, like using `isLOA1` rather than `isLOA3`. I think we are often revisiting those decisions, but it just gives me pause when I see them without related affirmative words about such a change. 

I made some minor comments about word choice, such as the difference between "level of assurance" an "account verification".   Just made some suggestions which you are free to take or leave.

File: src/applications/mhv-landing-page/constants.js [hunk 2]
```diff
@@ -1,6 +1,16 @@
 const { entries, freeze, values } = Object;
 
-export const HEALTH_TOOL_HEADINGS = freeze({
+// Please, keep these feature toggle settings up-to-date with production's feature toggles settings.
+const APPLICATION_FEATURE_TOGGLES = freeze({
```
[the remark this answers is not in the document]
Comment (radavis) [reply src/applications/mhv-landing-page/constants.js:4]: Carlos mentioned this as a possible improvement, as well, so I'll go ahead and this into the feature toggles mock.

File: src/applications/mhv-landing-page/tests/e2e/alerts/unregistered.cypress.spec.js (hunk 1 above)
Comment (radavis) [reply src/applications/mhv-landing-page/tests/e2e/alerts/unregistered.cypress.spec.js:11]: It counts associated VA treatment facilities to determine if they are registered as a VA Patient. Same difference, different avenue

File: src/applications/mhv-landing-page/tests/e2e/unread-sm-message-indicator/unread-sm-message-indicator.cypress.spec.js [hunk 3]
```diff
@@ -4,37 +4,35 @@ import ApiInitializer from '../utilities/ApiInitializer';
 import LandingPage from '../pages/LandingPage';
 
 describe(manifest.appName, () => {
-  describe('show indicator when there are unread messages', () => {
+  describe('the dot', () => {
```
[the remark this answers is not in the document]
Comment (jzucadi) [reply src/applications/mhv-landing-page/tests/e2e/unread-sm-message-indicator/unread-sm-message-indicator.cypress.spec.js:7]: same thought

**Review by jzucadi — DISMISSED** [review]

**Review by jzucadi — APPROVED** [review]

**Pull request merged**